From 0485ea89d64d0c42521de7e55715e80e57dc7d40 Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Fri, 26 Jun 2020 18:11:49 +0100
Subject: [PATCH 247/552] media: i2c: imx290: Switch set_hmax to use
 imx290_write_buffered_reg

imx290_set_hmax was using two independent writes to set up hmax,
when all other multi-register writes were using imx290_write_buffered_reg
which claims the group hold first.

Switch imx290_set_hmax to using imx290_write_buffered_reg too.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
---
 drivers/media/i2c/imx290.c | 15 ++++-----------
 1 file changed, 4 insertions(+), 11 deletions(-)

diff --git a/drivers/media/i2c/imx290.c b/drivers/media/i2c/imx290.c
index aeb95161b4fe..e64124a521b2 100644
--- a/drivers/media/i2c/imx290.c
+++ b/drivers/media/i2c/imx290.c
@@ -568,19 +568,12 @@ static int imx290_set_hmax(struct imx290 *imx290, u32 val)
 	u32 hmax = val + imx290->current_mode->width;
 	int ret;
 
-	ret = imx290_write_reg(imx290, IMX290_HMAX_LOW, (hmax & 0xff));
-	if (ret) {
-		dev_err(imx290->dev, "Error setting HMAX register\n");
-		return ret;
-	}
-
-	ret = imx290_write_reg(imx290, IMX290_HMAX_HIGH, ((hmax >> 8) & 0xff));
-	if (ret) {
+	ret = imx290_write_buffered_reg(imx290, IMX290_HMAX_LOW, 2,
+					hmax);
+	if (ret)
 		dev_err(imx290->dev, "Error setting HMAX register\n");
-		return ret;
-	}
 
-	return 0;
+	return ret;
 }
 
 static int imx290_set_vmax(struct imx290 *imx290, u32 val)
-- 
2.33.0

